### PR TITLE
Fixes for issues 161, 162, 163, 164, 165 and 166

### PR DESCRIPTION
**Pull request by Shane32**
## Related issue
This pull request relates to issues #161 #162 #163 #164 #165 and #166 

## Changes
- Updated package references, particularly, taking Discord.Net from beta to 2.0.1
- Removed package reference to Lamar and instead lay out program.cs similar to the design pattern used in ASP.Net Core for dependency injection
- Converted many classes to use dependency injection and/or not be global (static) classes.  Have not got them all though.  A few design flaws prevented perfectly clean code - see GlobalGuildAccount.Modify for example.
- Implemented a way to gracefully exit the application - just press Q in the console window
- Removed BotSettings.prefix as it is not being used

## Expected feedback
Although I do not expect any issues, a full test of all of this bot's features should be performed.  I just made sure the bot compiles, runs, and says hello.

## Comments
Thank you for an excellent template from which to start a new bot

### CODE REVIEW

**Comment by Shane32** [comment]
None of the remaining code review items by Codacy are possible to implement

File: CommunityBot/Features/GlobalAccounts/IGlobalUserAccounts.cs [hunk 1]
```diff
@@ -5,9 +5,8 @@
 
 namespace CommunityBot.Features.GlobalAccounts
 {
-    public interface IGlobalUserAccountProvider
+    public interface IGlobalUserAccounts : IGlobalAccounts
     {
-        GlobalUserAccount GetById(ulong userId);
```
Comment (petrspelos) [inline CommunityBot/Features/GlobalAccounts/IGlobalUserAccounts.cs]: The reason we went for `GetById` and not `GetUserAccount` is:
- When the method is used, `globalUserAccounts.GetUserAccount(id);` feels a bit redundant. You kinda mention user accounts twice. Moreover, `GetById` allows for different methods of getting accounts while stating the needed arguments clearly. :slightly_smiling_face:

**Comment by Shane32** [comment]
Ahh. I just changed the interface method name to match the class method name, without really reading what it was doing. GetById does sound more clear.  I’d suggest changing the class method name as well so they still match. Ditto for the save function if the alternate method name is preferred. I can push the change when I get home.

**Comment by Shane32** [comment]
I didn’t realize you had a newer code base when I started yesterday, and so I just thought I’d clear up a few of these design issues before starting my bot. I will certainly take a look at your new template. Thanks!

**Comment by Shane32** [comment]
So if `GetUserAccount(ulong id)` is changed to `GetById(ulong id)` then should `GetUserAccount(IUser user)` be changed?  To what?

**Comment by petrspelos** [comment]
> So if `GetUserAccount(ulong id)` is changed to `GetById(ulong id)` then should `GetUserAccount(IUser user)` be changed? To what?

I'd say perhaps something like `GetFromDiscordUser` although that seems a bit long. :thinking:

**Comment by Shane32** [comment]
Let me know if you'd like me to make any other changes to this pull request @petrspelos

File: CommunityBot/Configuration/BotSettings.cs [hunk 2]
```diff
@@ -62,5 +52,7 @@ private static ActionResult SaveSettings()
 
             return result;
         }
+        */
```
[the remark this answers is not in the document]
Comment (Shane32) [reply CommunityBot/Configuration/BotSettings.cs]: Done

**Pull request merged**